src/Decompiler/Typing/UnionAlternativeChooser.cs

```diff
@@ -0,0 +1,185 @@
+#region License
+/* 
+ * Copyright (C) 1999-2023 Pavel Tomin.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+
+using Reko.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Reko.Typing
+{
+    public class UnionAlternativeChooser : IDataTypeVisitor<bool>
```

Comment:
Please add a comment explaining the purpose of the class. The class name states that chooses an alternative for a union, but it would be helpful to have a high-level description of how the alternative is chosen.

Reply:
Done in dd919b1